### DISCUSSION TITLE
GTFL - A Graphical Terminal For Lisp - Web based presentation.

### DISCUSSION THREAD

**Discussion by aykaramba** [rabbibotton/clog#358] [General]
Rabbi Botton, I think you might find this interesting:

Link to Hacker News thread: https://news.ycombinator.com/item?id=40494097

Link to project: http://www.martin-loetzsch.de/gtfl/

It's interesting given that it is presented as an html page via hunchentoot.

**Comment by aykaramba** [comment]
Here is are sample applications of this tool: 

http://www.martin-loetzsch.de/gtfl/application-example-1/
http://www.martin-loetzsch.de/gtfl/application-example-2/

Click on any of those boxes.

**Comment by rabbibotton** [comment]
I saw it earlier, the main thing of the project is just an ajax pipe like clog-connect
The actual diagram part should not take much to adopt directly if desired and going to look more into.
The actual part of that that peaks my interest is not actually part of the code there

![image](https://github.com/rabbibotton/clog/assets/11616789/c38104ed-9b77-4c8e-8960-c6c4968431ac)

I want the code that does that graphing :)

**Comment by khinsen** [comment]
Somewhat related: has anyone played with graphviz (the JavaScript version at https://viz-js.com/) in CLOG? I guess it could be turned into a plugin.

**Comment by aykaramba** [comment]
That is so great, love that.  Nice find.

**Comment by rabbibotton** [comment]
https://github.com/mdaines/viz-js/tree/v3/packages%2Fexamples%2Fscript-tag

You would use this version so do not need to use Node.js and the binding
simple from what I see.

Ill see if can do something soon at least a basic binding.

**Comment by khinsen** [comment]
Viz.js works just fine with CLOG !

![graphviz-in-clog](https://github.com/user-attachments/assets/b64b4c1a-8b17-4ce9-9d43-5b759d045af3)

**Reply by rabbibotton** [reply]
Awesome, can you post a short snippet of what you did to load and use?

**Reply by khinsen** [reply]
Preparation:
 - load `viz-standalone.js`
 - load the following JS code

```
processGraphvizCodeElements = function(element) {
    Viz.instance().then(function(viz) {
        var graphvizCodeElements = element.getElementsByTagName('graphviz-code');
        for(var i = 0; i < graphvizCodeElements.length; ++ i)
        {
            var e = graphvizCodeElements[i];
            var svg = viz.renderSVGElement(e.textContent);
            e.textContent = '';
            e.appendChild(svg);
        }
    });
};
```

Then create HTML5 custom elements with tag `<graphviz-code>` that contain the text of a dot file (escaped of course).

Then run `processGraphvizCodeElements(window.currentInspectorView)`.

That relies on a variable `currentInspectorView` which I set before rendering each of my inspector views, but any other method of passing in a parent element of the graphviz stuff is fine as well.

**Reply by khinsen** [reply]
It's all in this commit: https://codeberg.org/khinsen/html-inspector-views/commit/6c7e4ed8c962a546ce42553c7fc029b935541c67
